protocol/messenger_mailserver.go

```diff
@@ -884,7 +899,11 @@ func (m *Messenger) processMailserverBatch(ms mailservers.Mailserver, batch Mail
 }
 
 func (m *Messenger) processMailserverBatchWithOptions(ms mailservers.Mailserver, batch MailserverBatch, pageLimit uint32, shouldProcessNextPage func(int) (bool, uint32), processEnvelopes bool) error {
```

Comment:
Would it be better to check for `canSyncWithStoreNodes` before `performMailserverRequest` is performed? Or not really? Because otherwise why should we perform the request, build the batch and then ignore it?

https://github.com/status-im/status-go/blob/9a59d6a4597165c075b2fa916f002343b7ff3a68/protocol/messenger_store_node_request_manager.go#L540-L550

Reply:
I have added the check in a bunch of more places, the code is a bit difficult to follow and there's many places where we need to check and make sure that we don't set the requests as done, I think I will just get it working so I can test it works, write some tests to cover the behaviour, and then try to refactor a bit (before merging), and retest again :) I will put it to draft since I expect a few changes